test/e2e/cluster_lifecycle_test.go

```diff
@@ -0,0 +1,261 @@
+
+			// We don't want to fail the test here if cleanup fails, but we should log it.
+			if err != nil {
+				GinkgoLogr.Error(err, "failed to start cluster deletion during cleanup")
+				return
+			}
+
+			_, err = poller.PollUntilDone(cleanupCtx, nil)
+			if err != nil {
+				GinkgoLogr.Error(err, "failed to poll for cluster deletion during cleanup")
+			}
+			By("Cleaning up the resource group")
+			// Use the framework's test context for resource group cleanup since it doesn't need the systemData policy
+			testCtx = framework.NewInvocationContext()
+			armResourcesClientFactory = testCtx.(interface {
+				GetARMResourcesClientFactoryOrDie(context.Context) *armresources.ClientFactory
+			}).GetARMResourcesClientFactoryOrDie(ctx)
+			resourceGroupClient = armResourcesClientFactory.NewResourceGroupsClient()
+			err = framework.DeleteResourceGroup(ctx, resourceGroupClient, resourceGroup, 60*time.Minute)
+			Expect(err).NotTo(HaveOccurred(), "failed to delete resource group during cleanup")
+			By("Verifying the resource group is deleted")
+			_, err = resourceGroupClient.Get(ctx, resourceGroup, nil)
+			Expect(err).ToNot(BeNil())
+			errMessage := fmt.Sprintf("The resource group '%s' could not be found.", resourceGroup)
+			Expect(err.Error()).To(ContainSubstring(errMessage))
+		}()
+
+		By("Create a new resource group for the cluster")
+		// Use the framework's test context for resource group operations since they don't need the systemData policy
+		testCtx = framework.NewInvocationContext()
+		armResourcesClientFactory = testCtx.(interface {
+			GetARMResourcesClientFactoryOrDie(context.Context) *armresources.ClientFactory
+		}).GetARMResourcesClientFactoryOrDie(ctx)
+		resourceGroupClient = armResourcesClientFactory.NewResourceGroupsClient()
+		resourceGroup = fmt.Sprintf("e2e-lifecycle-%s-rg", uuid.NewString()[:4])
+		_, err := framework.CreateResourceGroup(ctx, resourceGroupClient, resourceGroup, testCtx.(interface{ Location() string }).Location(), 10*time.Minute)
+		Expect(err).NotTo(HaveOccurred(), "failed to create resource group")
+
+		By("Deploying the infrastructure only bicep template")
+		deploymentName := fmt.Sprintf("e2e-lifecycle-%s-deployment", uuid.NewString()[:4])
+
+		// Read the bicep template file from the correct location
+		bicepTemplatePath := "../../e2e-setup/bicep/infra-only.bicep"
+		bicepContent, err := os.ReadFile(bicepTemplatePath)
+		Expect(err).NotTo(HaveOccurred(), "failed to read bicep template")
+
+		// Create deployment parameters for infra-only.bicep
+		deploymentParameters := map[string]string{
+			"clusterName":     clusterName,
+			"persistTagValue": "false",
+		}
+
+		// Use the framework's CreateBicepTemplateAndWait function for consistency
+		deploymentResult, err := framework.CreateBicepTemplateAndWait(
+			ctx,
+			armResourcesClientFactory.NewDeploymentsClient(),
+			resourceGroup,
+			deploymentName,
+			bicepContent,
```

Comment:
this part is a little tricky.  we can't use bicep directly, it  needs to be compiled first. Try converting in this file: https://github.com/Azure/ARO-HCP/blob/aa885890cfff7d2c717ddb1d964526659256e0d8/test/hack/update-bicep-json.sh#L55

You choose a source bicep file and a destination json file.  If you place your destination in `${OUTPUT_DIR}/<subdir>/foo.json`, then you can access it via the embedded filesystem like this: `framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/standard-cluster-create/customer-infra.json")),`